Src/FluentAssertions/Xml/XDocumentAssertions.cs

```diff
@@ -116,6 +118,44 @@ public AndConstraint<XDocumentAssertions> NotBeEquivalentTo(XDocument unexpected
             return new AndConstraint<XDocumentAssertions>(this);
         }
 
+        /// <summary>
+        /// Asserts that document is valid by binary represented XSD schema.
+        /// </summary>
+        /// <param name="xsd">XDS schema.</param>
+        /// <param name="because">
+        /// A formatted phrase as is supported by <see cref="string.Format(string,object[])" /> explaining why the assertion
+        /// is needed. If the phrase does not start with the word <i>because</i>, it is prepended automatically.
+        /// </param>
+        /// <param name="becauseArgs">
+        /// Zero or more objects to format using the placeholders in <paramref name="because" />.
+        /// </param>
+        /// <returns>Instance of <see cref="AndConstraint{T}"/>.</returns>
+        public AndConstraint<XDocumentAssertions> BeValidByXsd(byte[] xsd, string because = "", params object[] becauseArgs)
```

Comment:
Why do you expect the XSD in binary format?
Just a byte array is very unspecific. It is unclear what kind of "binary represented XSD schema" is expected.
I strongly recommend to accept here `string` or maybe `XDocument` only.